monkey/monkey_island/cc/ui/src/components/ui-components/HideInput.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import {InputGroup, FormControl} from 'react-bootstrap';
+import '../../styles/components/HideInput.scss'
+
+class HideInput extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      hidden: false
+    };
+    this.toggleShow = this.toggleShow.bind(this);
+  }
+
+  toggleShow() {
```

Comment:
My javascript is rusty, but if you define this like 
```suggestion
  toggleShow = () => {
```
then you can remove the `bind()` on line 12.